middleware/ratelimiter/store.go

```diff
@@ -0,0 +1,109 @@
+package ratelimiter
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/System-Glitch/goyave/v3"
+)
+
+type limiter struct {
+	requestQuota  int
+	counter       int
+	quotaDuration time.Duration
+	resetsAt      time.Time
+	mx            sync.RWMutex
```

Comment:
```suggestion
	mx            sync.Mutex
```
Only `newLimiter` uses the mutex now, and it both reads and write. Therefore, we can use a regular mutex.

Reply:
Oh ok... noted. Thank you